mlflow/protos/service.proto

```diff
@@ -2753,6 +2772,118 @@ message CalculateTraceFilterCorrelation {
   }
 }
 
+// Aggregation type for metrics.
+enum AggregationType {
+  // Count of entities.
+  COUNT = 1;
+  // Sum of values.
+  SUM = 2;
+  // Average of values.
+  AVG = 3;
+  // 50th percentile (median).
+  P50 = 4;
+  // 75th percentile.
+  P75 = 5;
+  // 90th percentile.
+  P90 = 6;
+  // 95th percentile.
+  P95 = 7;
+  // 99th percentile.
+  P99 = 8;
+}
+
+// Time granularity for time-based aggregation.
+enum TimeGranularity {
+  // Group by minute.
+  MINUTE = 1;
+  // Group by hour.
+  HOUR = 2;
+  // Group by day.
+  DAY = 3;
+  // Group by week.
+  WEEK = 4;
+  // Group by month.
+  MONTH = 5;
+}
+
+// View type for metrics aggregation.
+enum MetricsViewType {
+  // Aggregate at trace level.
+  TRACES = 1;
+  // Aggregate at span level.
+  SPANS = 2;
+  // Aggregate at assessment level.
+  ASSESSMENTS = 3;
+}
+
+// Query aggregated metrics for traces, spans, or assessments.
+message QueryTraceMetrics {
+  option (scalapb.message).extends = "com.databricks.rpc.RPC[$this.Response]";
+
+  // Required: The experiment IDs to search traces.
+  repeated string experiment_ids = 1;
+
+  // Required: The level at which to aggregate metrics.
+  optional MetricsViewType view_type = 2;
+
+  // Required: The metric_name to query (e.g., "trace", "latency", "token_count").
+  optional string metric_name = 3;
+
+  // Required: The aggregation types to apply.
+  repeated AggregationType aggregation_types = 4;
+
+  // Optional: Dimensions to group metrics by.
+  repeated string dimensions = 5;
+
+  // Optional: Filter expressions to apply.
+  repeated string filters = 6;
+
+  // Optional: Time granularity for grouping.
+  // When set, results automatically include a time dimension grouped by
+  // the specified granularity (e.g., DAY, HOUR).
+  optional TimeGranularity time_granularity = 7;
+
+  // Optional: Start of time range in milliseconds since epoch.
+  // Required if time_granularity is set.
+  optional int64 start_time_ms = 8;
+
+  // Optional: End of time range in milliseconds since epoch.
+  // Required if time_granularity is set.
+  optional int64 end_time_ms = 9;
+
+  // Optional: Maximum number of data points to return.
+  // Default: 1000
+  optional int32 max_results = 10 [default = 1000];
+
+  // Optional: Pagination token for fetching the next page of results.
+  optional string page_token = 11;
+
+  message Response {
+    // Data points grouped by dimensions.
+    repeated MetricDataPoint data_points = 1;
+
+    // Pagination token for fetching the next page.
+    // Empty if no more results are available.
+    optional string next_page_token = 2;
+  }
+}
+
+// A single data point with dimension values and metric values.
+message MetricDataPoint {
+  // Dimension values for this data point
+  // Keys correspond to dimensions
+  // e.g., {"trace_status": "OK", "time_bucket": "2025-11-13"}
```

Comment:
I'd recommend using unix time for "time_bucket" value. JSON does not support date type and it's much easier and flexible to format the time value in Frontend.